Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -390,20 +390,32 @@ def bottle_formula(f)
       mismatches = [:root_url, :prefix, :cellar, :rebuild].reject do |key|
         old_spec.send(key) == bottle.send(key)
       end
-      mismatches.delete(:cellar) if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
+      if old_spec.cellar == :any && bottle.cellar == :any_skip_relocation
+        mismatches.delete(:cellar)
+        bottle.cellar :any if OS.linux?
+      end
       unless mismatches.empty?
-        bottle_path.unlink if bottle_path.exist?
+        if bottle.prefix == "/home/linuxbrew/.linuxbrew"
+          bottle.cellar old_spec.cellar
+          bottle.prefix old_spec.prefix
+          opoo <<~EOS
```

Comment:
You're turning an error into a warning for no apparent reason.

Reply:
There are various macOS bottles that we also consider to be manually relocatable but aren't detected as such. It's not acceptable to me to simply override this. Additionally, you've provided a counterpoint which implies there may well be more: how will these ever be detected? If Linuxbrew wishes to claim bottles are relocatable which are not detected as such this logic should live in (in descending order of preference) CI scripts, a new external command or `brew test-bot`. A particular Linuxbrew bottling decision like this should not result in incorrect/inconsistent bottling on other Linux systems. I'm also concerned about behaviour changes like this affecting Homebrew's bottling some of which is run on Linux.

Reply:
The example I gave, `curl`, is not relocatable on Linux, because it's built using `openssl` on Linux, which causes it not to be relocatable. `curl-openssl` is also similarly not relocatable on macOS. That's the only exception I have run into. A tool that is relocatable on macOS is also relocatable on Linux, empirically true 99.92% of the time (1289/1290).

Reply:
We've talked about this before in the context of relocatable bottles and I'm not convinced that things that simply pass `test do` when relocated have 100% of their functionality working perfectly when relocated. Again, that's a debate about improving the behaviour of relocation rather than simply overriding it in this case.